middleware/permissions.js

```diff
@@ -1,21 +1,23 @@
 // Copyright (c) Amazon.com, Inc. or its affiliates and others. Licensed under the MIT license.
 // SPDX-License-Identifier: MIT
+const asyncMiddleware = require('./asyncMiddleware')
 
 /**
  * Middleware that checks for permissions for this request based on team membership.
  *
  * Usage: `app.get('/some/route', permissionsCheck('harvesters'), (req, res) => ...)`
  */
 function permissionsCheck(permission) {
-  return (request, response, next) => {
-    const userTeams = request.app.locals.user.github.teams
+  return asyncMiddleware(async function permissionsCheck(request, response, next) {
+    const userTeams = await request.app.locals.user.github.getTeams()
     const requiredTeams = permissions[permission]
-    const intersection = requiredTeams.filter(t => userTeams.includes(t))
-    if (requiredTeams.length === 0 || intersection.length > 0) return next()
-    const error = new Error(`No permission to '${permission}' (needs team membership)`)
-    error.status = 401
-    next(error)
-  }
+    const intersection = requiredTeams.filter(t => (userTeams || []).includes(t))
+    if (requiredTeams.length !== 0 && intersection.length == 0) {
+      return response.status(401).send(`No permission to '${permission}' (needs team membership)`)
+    }
+
+    next()
```

Comment:
next(error) would bubble up and return the error as expected. I would refrain from changing these lines.

Reply:
@geneh  We do this in some of our other routes-- is it wrong there?

Reply:
It's not completely wrong, just cleaner this way. Feel free to either change to the previous code or merge.